test/formal/and_distributed_over_shl.py

```diff
@@ -0,0 +1,25 @@
+from opcodes import AND, SHL
+from rule import Rule
+from z3 import BitVec
+
+"""
+Rule:
+AND(SHL(A,X), SHL(A,Y)) -> SHL(A, AND(X,Y))
```

Comment:
Just for clarity: Please update to also use X, Y, Z here.

Reply:
Done